### PR TITLE
[MRG] add an `lca gather` signature decomposition approach

### LINKED ISSUE #387: `sourmash lca classify` db error message isn't very friendly
I recently ran `sourmash lca classify` with a corrupted database inside of a snakemake workflow, producing the following error message:

```
rule run_sourmash_LCA_sylv:
    input: inputs/genbank_lca/genbank-k31.lca.json, outputs/sylvestris/Olea_europaea_1kb_scaffolds-k31.fa.sig
    output: outputs/sourmash_lca/Olea_europaea_1kb_scaffolds-k31.lca.txt
    jobid: 1

Activating conda environment /home/ubuntu/olive_genome/dummy/.snakemake/conda/54f9bee7.
Traceback (most recent call last):k_lca/genbank-k31.lca.json
  File "/home/ubuntu/olive_genome/dummy/.snakemake/conda/54f9bee7/bin/sourmash", line 11, in <module>
    load_entry_point('sourmash', 'console_scripts', 'sourmash')()
  File "/home/ubuntu/olive_genome/dummy/src/sourmash/sourmash_lib/__main__.py", line 63, in main
    cmd(sys.argv[2:])
  File "/home/ubuntu/olive_genome/dummy/src/sourmash/sourmash_lib/lca/__main__.py", line 46, in main
    cmd(sysv_args[1:])
  File "/home/ubuntu/olive_genome/dummy/src/sourmash/sourmash_lib/lca/command_classify.py", line 111, in classify
    dblist, ksize, scaled = lca_utils.load_databases(args.db, args.scaled)
  File "/home/ubuntu/olive_genome/dummy/src/sourmash/sourmash_lib/lca/lca_utils.py", line 249, in load_databases
    lca_db.load(db_name)
  File "/home/ubuntu/olive_genome/dummy/src/sourmash/sourmash_lib/lca/lca_utils.py", line 139, in load
    load_d = json.load(fp)
  File "/home/ubuntu/olive_genome/dummy/.snakemake/conda/54f9bee7/lib/python3.6/json/__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/home/ubuntu/olive_genome/dummy/.snakemake/conda/54f9bee7/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/home/ubuntu/olive_genome/dummy/.snakemake/conda/54f9bee7/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/ubuntu/olive_genome/dummy/.snakemake/conda/54f9bee7/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
Error in rule run_sourmash_LCA_sylv:
    jobid: 1
    output: outputs/sourmash_lca/Olea_europaea_1kb_scaffolds-k31.lca.txt

RuleException:
CalledProcessError in line 59 of /home/ubuntu/olive_genome/dummy/sourmashlca.snakefile:
Command 'source activate /home/ubuntu/olive_genome/dummy/.snakemake/conda/54f9bee7; set -euo pipefail;  
    sourmash lca classify --db inputs/genbank_lca/genbank-k31.lca.json --query outputs/sylvestris/Olea_europaea_1kb_scaffolds-k31.fa.sig -o outputs/sourmash_lca/Olea_europaea_1kb_scaffolds-k31.lca.txt ' returned non-zero exit status 1.
  File "/home/ubuntu/olive_genome/dummy/sourmashlca.snakefile", line 59, in __rule_run_sourmash_LCA_sylv
  File "/home/ubuntu/miniconda3/envs/snakemake/lib/python3.6/concurrent/futures/thread.py", line 55, in run
```

As @titus indicated, this is from a corrupted database. However, this error message wasn't very friendly. A more informative/friendly error message would be helpful!

### LINKED ISSUE #401: update LCA DB format to have additional information
it would be nice to have access to the original sequence names; we should also support strain resolution by default (if provided).

### PR DESCRIPTION
**Pull request by ctb**
Ref #384. Unlike 'sourmash gather' this *requires* taxonomy information from an LCA database.

It is also
* blazingly fast
* quite memory intensive

Unlike `sourmash gather` this will not print out arbitrary genome names or anything - it's solely about taxonomic display.  It also alerts the user to the number of equivalent matches and outputs it to the CSV (but only displays one match).

other additions:

- [x] LCA commands now include strain information (and leave it empty if it's not in the database)
- [x] LCA databases now contain a 'signatures_to_name' mapping so we can retrieve the name of the signature from which hashes come
- [x] fixes a bug in `sourmash gather --output-unassigned`
- [x] fixes a display problem with `sourmash lca` subcommands where subcommand was not output to help
- [x] fixes #387 - fail more nicely when failing to parse LCA DBs
- [x] fixes #401 - adds sequence names and strain rez

checklist:

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
Latest output; compare with regular gather results at: https://sourmash.readthedocs.io/en/latest/tutorials.html#what-s-in-my-metagenome

```
% sourmash lca gather shakya-unaligned-contigs.sig genbank-k31.lca.json.gz
loaded 1 LCA databases. ksize=31, scaled=10000
loaded query: mqc500.QC.AMBIGUOUS.99.unalign... (k=31)

overlap     p_query
---------   -------
1.8 Mbp      14.6%      Fusobacterium nucleatum
1.0 Mbp       7.8%      Proteiniclasticum ruminis
1.0 Mbp       7.7%      Haloferax volcanii
0.9 Mbp       7.4%      Nostoc sp. PCC 7120
0.9 Mbp       7.0%      Shewanella baltica
0.8 Mbp       6.0%      Desulfovibrio vulgaris
0.6 Mbp       4.9%      Thermus thermophilus
0.6 Mbp       4.4%      Ruegeria pomeroyi
480.0 kbp     3.8%      Herpetosiphon aurantiacus
410.0 kbp     3.3%      Sulfitobacter sp. NAS-14.1
150.0 kbp     1.2%      Deinococcus radiodurans
150.0 kbp     1.2%      Thermotoga sp. RQ2
140.0 kbp     1.1%      Sulfitobacter sp. EE-36
130.0 kbp     1.0%      Streptococcus agalactiae
100.0 kbp     0.8%      Salinispora arenicola
100.0 kbp     0.8%      Fusobacterium sp. OBRC1
60.0 kbp      0.5%      Paraburkholderia xenovorans
50.0 kbp      0.4%      Bacteroides vulgatus
50.0 kbp      0.4%      Methanocaldococcus jannaschii
50.0 kbp      0.4%      Sulfurihydrogenibium sp. YO3AOP1
30.0 kbp      0.2%      Zymomonas mobilis
30.0 kbp      0.2%      Persephonella marina
30.0 kbp      0.2%      Fusobacterium hwasookii
30.0 kbp      0.2%      Pseudomonas aeruginosa
20.0 kbp      0.2%      Ruminiclostridium thermocellum
20.0 kbp      0.2%      Chlorobium phaeobacteroides
```

**Comment by ctb** [comment]
Latest output; compare with regular gather results at: https://sourmash.readthedocs.io/en/latest/tutorials.html#what-s-in-my-metagenome

```
% sourmash lca gather shakya-unaligned-contigs.sig genbank-k31.lca.json.gz
loaded 1 LCA databases. ksize=31, scaled=10000
loaded query: mqc500.QC.AMBIGUOUS.99.unalign... (k=31)

overlap     p_query p_match
---------   ------- --------
1.8 Mbp      14.6%    9.1%      Fusobacterium nucleatum
1.0 Mbp       7.8%   16.3%      Proteiniclasticum ruminis
1.0 Mbp       7.7%   25.9%      Haloferax volcanii
0.9 Mbp       7.4%   11.8%      Nostoc sp. PCC 7120
0.9 Mbp       7.0%    5.8%      Shewanella baltica
0.8 Mbp       6.0%    8.6%      Desulfovibrio vulgaris
0.6 Mbp       4.9%   12.6%      Thermus thermophilus
0.6 Mbp       4.4%   11.2%      Ruegeria pomeroyi
480.0 kbp     3.8%    7.6%      Herpetosiphon aurantiacus
410.0 kbp     3.3%   10.5%      Sulfitobacter sp. NAS-14.1
150.0 kbp     1.2%    4.5%      Deinococcus radiodurans (** 1 equal matches)
150.0 kbp     1.2%    8.2%      Thermotoga sp. RQ2
140.0 kbp     1.1%    4.1%      Sulfitobacter sp. EE-36
130.0 kbp     1.0%    0.4%      Enterococcus faecalis (** 1 equal matches)
100.0 kbp     0.8%    0.3%      Salinispora arenicola (** 1 equal matches)
100.0 kbp     0.8%    4.2%      Fusobacterium sp. OBRC1
60.0 kbp      0.5%    0.7%      Paraburkholderia xenovorans
50.0 kbp      0.4%    0.3%      Bacteroides vulgatus (** 2 equal matches)
50.0 kbp      0.4%    3.2%      Methanocaldococcus jannaschii (** 1 equal matches)
50.0 kbp      0.4%    2.6%      Sulfurihydrogenibium sp. YO3AOP1
30.0 kbp      0.2%    0.4%      Zymomonas mobilis (** 6 equal matches)
30.0 kbp      0.2%    1.6%      Persephonella marina (** 5 equal matches)
30.0 kbp      0.2%    0.0%      Escherichia coli (** 4 equal matches)
30.0 kbp      0.2%    0.7%      Fusobacterium hwasookii
20.0 kbp      0.2%    0.5%      Ruminiclostridium thermocellum (** 6 equal matches)
20.0 kbp      0.2%    0.4%      Chlorobium phaeobacteroides (** 5 equal matches)
20.0 kbp      0.2%    0.8%      Fusobacterium sp. HMSC064B11 (** 4 equal matches)
20.0 kbp      0.2%    0.7%      Caldicellulosiruptor bescii (** 3 equal matches)
20.0 kbp      0.2%    1.1%      Sulfurihydrogenibium yellowstonense (** 2 equal matches)
20.0 kbp      0.2%    1.0%      Streptococcus sp. HMSC074F05 (** 1 equal matches)
10.0 kbp      0.1%    0.3%      Geobacter anodireducens (** 347 equal matches)
10.0 kbp      0.1%    0.5%      Thermus kawarayensis (** 345 equal matches)
10.0 kbp      0.1%    0.5%      Fusobacterium sp. HMSC065F01 (** 344 equal matches)
10.0 kbp      0.1%    0.5%      Pyrococcus furiosus (** 343 equal matches)
10.0 kbp      0.1%    0.2%      Desulfovibrio piger (** 342 equal matches)
10.0 kbp      0.1%    0.1%      Desulfovibrio desulfuricans (** 341 equal matches)
10.0 kbp      0.1%    0.5%      Aciduliprofundum boonei (** 340 equal matches)
10.0 kbp      0.1%    0.2%      Desulfovibrio sp. A2 (** 339 equal matches)
10.0 kbp      0.1%    0.5%      Thermotoga sp. RQ7 (** 338 equal matches)
10.0 kbp      0.1%    0.1%      Achromobacter sp. 2789STDY5608624 (** 336 equal matches)
10.0 kbp      0.1%    0.3%      Nitrosomonas europaea (** 300 equal matches)
10.0 kbp      0.1%    0.2%      Acidovorax sp. RAC01 (** 299 equal matches)
10.0 kbp      0.1%    0.5%      Thermus amyloliquefaciens (** 4 equal matches)
10.0 kbp      0.1%    0.3%      Desulfocurvus vexinensis (** 3 equal matches)
10.0 kbp      0.1%    0.4%      Fusobacterium sp. CM22 (** 1 equal matches)
10.0 kbp      0.1%    0.5%      Wolinella succinogenes

22.1% (2.8 Mbp) have no assignment.

```

**Comment by ctb** [comment]
I think this is ready for review @brooksph @taylorreiter.

**Comment by ctb** [comment]
The output CSV looks like this:
```
intersect_bp,f_match,f_unique_to_query,f_unique_weighted,average_abund,name,n_equal_matches,superkingdom,phylum,class,order,family,genus,species,strain
1830000,0.09081885856079404,0.14558472553699284,0.14558472553699284,1.0,,0,Bacteria,Fusobacteria,Fusobacteriia,Fusobacteriales,Fusobacteriaceae,Fusobacterium,Fusobacterium nucleatum,
980000,0.16333333333333333,0.07796340493237867,0.07796340493237867,1.0,,0,Bacteria,Firmicutes,Clostridia,Clostridiales,Clostridiaceae,Proteiniclasticum,Proteiniclasticum ruminis,
970000,0.25935828877005346,0.07716785998408911,0.07716785998408911,1.0,,0,Archaea,Euryarchaeota,Halobacteria,Haloferacales,Haloferacaceae,Haloferax,Haloferax volcanii,
930000,0.11802030456852793,0.07398568019093078,0.07398568019093078,1.0,,0,Bacteria,Cyanobacteria,unassigned,Nostocales,Nostocaceae,Nostoc,Nostoc sp. PCC 7120,
880000,0.05751633986928104,0.07000795544948289,0.07000795544948289,1.0,,0,Bacteria,Proteobacteria,Gammaproteobacteria,Alteromonadales,Shewanellaceae,Shewanella,Shewanella baltica,
760000,0.08558558558558559,0.060461416070007955,0.060461416070007955,1.0,,0,Bacteria,Proteobacteria,Deltaproteobacteria,Desulfovibrionales,Desulfovibrionaceae,Desulfovibrio,Desulfovibrio vulgaris,
620000,0.12601626016260162,0.049323786793953855,0.049323786793953855,1.0,,0,Bacteria,Deinococcus-Thermus,Deinococci,Thermales,Thermaceae,Thermus,Thermus thermophilus,
550000,0.11178861788617886,0.04375497215592681,0.04375497215592681,1.0,,0,Bacteria,Proteobacteria,Alphaproteobacteria,Rhodobacterales,Rhodobacteraceae,Ruegeria,Ruegeria pomeroyi,
480000,0.0761904761904762,0.03818615751789976,0.03818615751789976,1.0,,0,Bacteria,Chloroflexi,Chloroflexia,Herpetosiphonales,Herpetosiphonaceae,Herpetosiphon,Herpetosiphon aurantiacus,
410000,0.10539845758354756,0.03261734287987271,0.03261734287987271,1.0,,0,Bacteria,Proteobacteria,Alphaproteobacteria,Rhodobacterales,Rhodobacteraceae,Sulfitobacter,Sulfitobacter sp. NAS-14.1,
150000,0.04477611940298507,0.011933174224343675,0.011933174224343675,1.0,,1,Bacteria,Deinococcus-Thermus,Deinococci,Deinococcales,Deinococcaceae,Deinococcus,Deinococcus radiodurans,
150000,0.08196721311475409,0.011933174224343675,0.011933174224343675,1.0,,0,Bacteria,Thermotogae,Thermotogae,Thermotogales,Thermotogaceae,Thermotoga,Thermotoga sp. RQ2,
140000,0.04057971014492753,0.011137629276054098,0.011137629276054098,1.0,,0,Bacteria,Proteobacteria,Alphaproteobacteria,Rhodobacterales,Rhodobacteraceae,Sulfitobacter,Sulfitobacter sp. EE-36,
130000,0.003914483589280337,0.010342084327764518,0.010342084327764518,1.0,,1,Bacteria,Firmicutes,Bacilli,Lactobacillales,Enterococcaceae,Enterococcus,Enterococcus faecalis,
100000,0.0032061558191728116,0.007955449482895784,0.007955449482895784,1.0,,1,Bacteria,Actinobacteria,Actinobacteria,Micromonosporales,Micromonosporaceae,Salinispora,Salinispora arenicola,
100000,0.04201680672268908,0.007955449482895784,0.007955449482895784,1.0,,0,Bacteria,Fusobacteria,Fusobacteriia,Fusobacteriales,Fusobacteriaceae,Fusobacterium,Fusobacterium sp. OBRC1,
60000,0.006521739130434782,0.00477326968973747,0.00477326968973747,1.0,,0,Bacteria,Proteobacteria,Betaproteobacteria,Burkholderiales,Burkholderiaceae,Paraburkholderia,Paraburkholderia xenovorans,
50000,0.003360215053763441,0.003977724741447892,0.003977724741447892,1.0,,2,Bacteria,Bacteroidetes,Bacteroidia,Bacteroidales,Bacteroidaceae,Bacteroides,Bacteroides vulgatus,
50000,0.03184713375796178,0.003977724741447892,0.003977724741447892,1.0,,1,Archaea,Euryarchaeota,Methanococci,Methanococcales,Methanocaldococcaceae,Methanocaldococcus,Methanocaldococcus jannaschii,
50000,0.026455026455026454,0.003977724741447892,0.003977724741447892,1.0,,0,Bacteria,Aquificae,Aquificae,Aquificales,Hydrogenothermaceae,Sulfurihydrogenibium,Sulfurihydrogenibium sp. YO3AOP1,
30000,0.004497751124437781,0.002386634844868735,0.002386634844868735,1.0,,6,Bacteria,Proteobacteria,Alphaproteobacteria,Sphingomonadales,Sphingomonadaceae,Zymomonas,Zymomonas mobilis,
30000,0.015789473684210527,0.002386634844868735,0.002386634844868735,1.0,,5,Bacteria,Aquificae,Aquificae,Aquificales,Hydrogenothermaceae,Persephonella,Persephonella marina,
30000,0.0002227336847575915,0.002386634844868735,0.002386634844868735,1.0,,4,Bacteria,Proteobacteria,Gammaproteobacteria,Enterobacterales,Enterobacteriaceae,Salmonella,Salmonella enterica,
30000,0.006896551724137931,0.002386634844868735,0.002386634844868735,1.0,,0,Bacteria,Fusobacteria,Fusobacteriia,Fusobacteriales,Fusobacteriaceae,Fusobacterium,Fusobacterium hwasookii,
20000,0.004640371229698376,0.0015910898965791568,0.0015910898965791568,1.0,,6,Bacteria,Firmicutes,Clostridia,Clostridiales,Ruminococcaceae,Ruminiclostridium,Ruminiclostridium thermocellum,
20000,0.0035714285714285713,0.0015910898965791568,0.0015910898965791568,1.0,,5,Bacteria,Chlorobi,Chlorobia,Chlorobiales,Chlorobiaceae,Chlorobium,Chlorobium phaeobacteroides,
20000,0.008064516129032258,0.0015910898965791568,0.0015910898965791568,1.0,,4,Bacteria,Fusobacteria,Fusobacteriia,Fusobacteriales,Fusobacteriaceae,Fusobacterium,Fusobacterium sp. HMSC064B11,
20000,0.006711409395973154,0.0015910898965791568,0.0015910898965791568,1.0,,3,Bacteria,Firmicutes,Clostridia,Thermoanaerobacterales,Thermoanaerobacterales Family III. Incertae Sedis,Caldicellulosiruptor,Caldicellulosiruptor bescii,
20000,0.01092896174863388,0.0015910898965791568,0.0015910898965791568,1.0,,2,Bacteria,Aquificae,Aquificae,Aquificales,Hydrogenothermaceae,Sulfurihydrogenibium,Sulfurihydrogenibium yellowstonense,
20000,0.00975609756097561,0.0015910898965791568,0.0015910898965791568,1.0,,1,Bacteria,Firmicutes,Bacilli,Lactobacillales,Streptococcaceae,Streptococcus,Streptococcus sp. HMSC074F05,
10000,0.002638522427440633,0.0007955449482895784,0.0007955449482895784,1.0,,347,Bacteria,Proteobacteria,Deltaproteobacteria,Desulfuromonadales,Geobacteraceae,Geobacter,Geobacter anodireducens,
10000,0.004878048780487805,0.0007955449482895784,0.0007955449482895784,1.0,,345,Bacteria,Deinococcus-Thermus,Deinococci,Thermales,Thermaceae,Thermus,Thermus kawarayensis,
10000,0.004975124378109453,0.0007955449482895784,0.0007955449482895784,1.0,,344,Bacteria,Fusobacteria,Fusobacteriia,Fusobacteriales,Fusobacteriaceae,Fusobacterium,Fusobacterium sp. HMSC065F01,
10000,0.005434782608695652,0.0007955449482895784,0.0007955449482895784,1.0,,343,Archaea,Euryarchaeota,Thermococci,Thermococcales,Thermococcaceae,Pyrococcus,Pyrococcus furiosus,
10000,0.0018281535648994515,0.0007955449482895784,0.0007955449482895784,1.0,,342,Bacteria,Proteobacteria,Deltaproteobacteria,Desulfovibrionales,Desulfovibrionaceae,Desulfovibrio,Desulfovibrio piger,
10000,0.0006807351940095302,0.0007955449482895784,0.0007955449482895784,1.0,,341,Bacteria,Proteobacteria,Deltaproteobacteria,Desulfovibrionales,Desulfovibrionaceae,Desulfovibrio,Desulfovibrio desulfuricans,
10000,0.004761904761904762,0.0007955449482895784,0.0007955449482895784,1.0,,340,Archaea,Euryarchaeota,unassigned,unassigned,unassigned,Aciduliprofundum,Aciduliprofundum boonei,
10000,0.0022172949002217295,0.0007955449482895784,0.0007955449482895784,1.0,,339,Bacteria,Proteobacteria,Deltaproteobacteria,Desulfovibrionales,Desulfovibrionaceae,Desulfovibrio,Desulfovibrio sp. A2,
10000,0.005025125628140704,0.0007955449482895784,0.0007955449482895784,1.0,,338,Bacteria,Thermotogae,Thermotogae,Thermotogales,Thermotogaceae,Thermotoga,Thermotoga neapolitana,
10000,0.0015267175572519084,0.0007955449482895784,0.0007955449482895784,1.0,,336,Bacteria,Proteobacteria,Betaproteobacteria,Burkholderiales,Alcaligenaceae,Achromobacter,Achromobacter sp. 2789STDY5608612,
10000,0.003278688524590164,0.0007955449482895784,0.0007955449482895784,1.0,,300,Bacteria,Proteobacteria,Betaproteobacteria,Nitrosomonadales,Nitrosomonadaceae,Nitrosomonas,Nitrosomonas europaea,
10000,0.0018726591760299626,0.0007955449482895784,0.0007955449482895784,1.0,,299,Bacteria,Proteobacteria,Betaproteobacteria,Burkholderiales,Comamonadaceae,Hydrogenophaga,Hydrogenophaga sp. A37,
10000,0.0049261083743842365,0.0007955449482895784,0.0007955449482895784,1.0,,4,Bacteria,Deinococcus-Thermus,Deinococci,Thermales,Thermaceae,Thermus,Thermus amyloliquefaciens,
10000,0.002881844380403458,0.0007955449482895784,0.0007955449482895784,1.0,,3,Bacteria,Proteobacteria,Deltaproteobacteria,Desulfovibrionales,Desulfovibrionaceae,Desulfocurvus,Desulfocurvus vexinensis,
10000,0.004081632653061225,0.0007955449482895784,0.0007955449482895784,1.0,,1,Bacteria,Fusobacteria,Fusobacteriia,Fusobacteriales,Fusobacteriaceae,Fusobacterium,Fusobacterium sp. CM22,
10000,0.004830917874396135,0.0007955449482895784,0.0007955449482895784,1.0,,0,Bacteria,Proteobacteria,Epsilonproteobacteria,Campylobacterales,Helicobacteraceae,Wolinella,Wolinella succinogenes,
```

**Comment by ctb** [comment]
(in order to get strain information in the output, we need to rebuild the genbank-k31.lca.json file with strain info in it - workin' on it!)

File: sourmash_lib/lca/command_compare_csv.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 from sourmash_lib.lca.command_index import load_taxonomy_assignments
 
 def compare_csv(args):
-    p = argparse.ArgumentParser()
+    p = argparse.ArgumentParser(prog="sourmash lca compare_csv")
```
[the remark this answers is not in the document]
Comment (luizirber) [reply sourmash_lib/lca/command_compare_csv.py]: Punted to #415

File: sourmash_lib/lca/command_gather.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+from collections import Counter, defaultdict, namedtuple
+
+import sourmash_lib
+from sourmash_lib import sourmash_args
+from sourmash_lib.logging import notify, error, print_results
+from sourmash_lib.lca import lca_utils
+from sourmash_lib.lca.lca_utils import debug, set_debug
+from sourmash_lib.search import format_bp
+
+LCAGatherResult = namedtuple('LCAGatherResult',
```
[the remark this answers is not in the document]
Comment (luizirber) [reply sourmash_lib/lca/command_gather.py]: Punted to #416

**Comment by ctb** [comment]
yay w00t!

**Pull request merged**